### PR TITLE
Add notify to Slack threads

### LINKED ISSUE #1048: Improve Slack Notification Error Handling
##### ISSUE TYPE
 - Enhancement

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
Currently we only surface errors when sending slack notification if a Python exception occurs.
We could do better if we read the event stream after our `rtm_send_message` call to see if there
were any error events waiting.

A common error is when the bot has not been invited to a channel that we are attempting to send the notification to. This can be caught in the message stream, but does not throw an exception.

##### ADDITIONAL INFORMATION

https://github.com/ansible/awx/blob/devel/awx/main/notifications/slack_backend.py#L49

### PR DESCRIPTION
**Pull request by walkafwalka**
Signed-off-by: walkafwalka <41709139+walkafwalka@users.noreply.github.com>

##### SUMMARY
Adds the ability to specify a Slack thread to reply to.

Fixes #1048 and #1741.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API
 - UI

##### AWX VERSION
```
awx: 1.0.6.44
```

##### ADDITIONAL INFORMATION
I also bumped the slackclient and added the Slack error message to error log.

### CODE REVIEW

**Comment by walkafwalka** [comment]
@matburt

**Review by matburt — COMMENTED** [review]
Looks good... can you add the thread-reply detail to the notifications architecture documents under docs/

**Comment by walkafwalka** [comment]
Aaaand, done.

**Comment by jakemcdermott** [comment]
hello @walkafwalka 

Thanks for the PR. You'll need to resolve merge conflicts with the devel branch  for this to be mergeable.

File: requirements/requirements.txt [hunk 1]
```diff
@@ -106,7 +106,6 @@ constantly==15.1.0        # via twisted
 cryptography==2.1.4       # via adal, azure-cosmosdb-table, azure-storage-common, pyopenssl, requests, secretstorage
 daphne==1.3.0
 decorator==4.2.1
-debtcollector==1.15.0     # via oslo.config, oslo.utils, python-designateclient, python-keystoneclient, python-neutronclient
```
Comment (jakemcdermott) [inline requirements/requirements.txt:109]: It is not clear to me why this dependency is being removed, and I would like to know more!

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

**Comment by matburt** [comment]
ping @walkafwalka can you address the concerns above?

**Comment by walkafwalka** [comment]
Sorry. I do not have time to address this fix for this project. Anyone feel free to pick this up.

**Comment by tanujdevops** [comment]
Hi @matburt, I would like to pick this issue up again. Can you please assign it to me?

**Comment by tanujdevops** [comment]
Hi @matburt, I would like to pick this issue up again. Can you please assign it to me?

**Pull request closed without merging**